app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,15 @@
+class PostsController < ApplicationController
+  before_action :redirect_if_document_not_found, only: :show
+
+  helper_method :document
+
+  private
+
+  def redirect_if_document_not_found
+    redirect_to not_found_path unless document
```

Comment:
This should use `not_found` from `ApplicationController` rather than redirecting:

```ruby
def show
  not_found unless document
end
```